src/nix.rs

```diff
@@ -1,13 +1,25 @@
 use crate::check_url;
 use anyhow::{Context, Result};
+use data_encoding::BASE64;
 use log::debug;
+use nix_compat::nixhash::from_str;
```

Comment:
Slight nit: Importing a function with such a generic name as `from_str` and that for just one usage; I'd prefer simply using the fully qualified name at call site instead.